kubespawner/spawner.py

```diff
@@ -1768,11 +1770,57 @@ def get_env(self):
         """
 
         env = super(KubeSpawner, self).get_env()
+        
+        # Filter env. variables with only values and pass it to secret
+        env = {k: v for k, v in (env or {}).items() if type(v) != dict}
         # deprecate image
         env['JUPYTER_IMAGE_SPEC'] = self.image
         env['JUPYTER_IMAGE'] = self.image
+        
+        if self.internal_ssl:
+            """
+            cert_paths:
+                certificate path references
+            hub_ca:
+                Path to the hub certificate authority
+            """
+            with open(self.cert_paths['keyfile'], 'r') as file:
+                env['ssl.key'] = file.read()
+
+            with open(self.cert_paths['certfile'], 'r') as file:
+                env['ssl.crt'] = file.read()
+
+            with open(self.cert_paths['cafile'], 'r') as file:
+                env["notebooks-ca_trust.crt"] = file.read()
+
+            with open(self.internal_trust_bundles['hub-ca'], 'r') as file:
+                env["notebooks-ca_trust.crt"] = env[
+                    "notebooks-ca_trust.crt"
+                ] + file.read()
+            env['JUPYTERHUB_SSL_KEYFILE'] = self.secret_mount_path + "ssl.key"
+            env['JUPYTERHUB_SSL_CERTFILE'] = self.secret_mount_path + "ssl.crt"
+            env['JUPYTERHUB_SSL_CLIENT_CA'] = (self.secret_mount_path + "notebooks-ca_trust.crt")
 
         return env
+    
+    def get_env_value_from(self):
```

Comment:
@consideRatio I've created a separate `get_env_value_from` to filter only the env. variables which has `value_from`, hence the name :). Now `get_env` will filter only the env. variables with values but not `value_from`. Also i'm using `V1EnvVar` in this since we already have the right format for it to process.

For example.  
This `"TEST_KEY_2": {'valueFrom': {'secretKeyRef': {'name': 'my-test-secret', 'key': 'password'}}}` being formatted as `{"name": "TEST_KEY_2", "value_from": {'secretKeyRef': {'name': 'my-test-secret', 'key': 'password'}}}`. I'm creating list of these env. variables containing value_from which are then passed to V1Container object in `env` variable.